src/System.Net.Ping/src/System/Net/NetworkInformation/Ping.Unix.cs

```diff
@@ -88,15 +88,27 @@ private SocketConfig GetSocketConfig(IPAddress address, byte[] buffer, int timeo
 
         private Socket GetRawSocket(SocketConfig socketConfig)
         {
+            IPEndPoint ep = socketConfig.EndPoint as IPEndPoint;
+
             // Setting Socket.DontFragment and .Ttl is not supported on Unix, so socketConfig.Options is ignored.
-            AddressFamily addrFamily = ((IPEndPoint)socketConfig.EndPoint).Address.AddressFamily;
+            AddressFamily addrFamily = ep.Address.AddressFamily;
```

Comment:
This change is now going to result in a NullReferenceException if the cast fails.  It'd be better to get an InvalidCastException if it were to ever fail, so the line above should be changed from:
```C#
IPEndPoint ep = socketConfig.EndPoint as IPEndPoint;
```
to:
```C#
IPEndPoint ep = (IPEndPoint)socketConfig.EndPoint;
```